sky/core.py

```diff
@@ -590,18 +591,22 @@ def cancel(
         sky_logging.print(f'{colorama.Fore.YELLOW}'
                           f'Cancelling all jobs on cluster {cluster_name!r}...'
                           f'{colorama.Style.RESET_ALL}')
-    elif not job_ids:
-        # job_ids is not None or empty.
+    elif job_ids is None:
+        # all = False, job_ids is None => cancel the latest running job.
         sky_logging.print(
             f'{colorama.Fore.YELLOW}'
             f'Cancelling latest running job on cluster {cluster_name!r}...'
             f'{colorama.Style.RESET_ALL}')
-    else:
+    elif len(job_ids):
+        # all = False, len(job_ids) > 0 => cancel the specified jobs.
         jobs_str = ', '.join(map(str, job_ids))
         sky_logging.print(
             f'{colorama.Fore.YELLOW}'
             f'Cancelling jobs ({jobs_str}) on cluster {cluster_name!r}...'
             f'{colorama.Style.RESET_ALL}')
+    else:
+        # all = False, len(job_ids) == 0 => no jobs to cancel.
```

Comment:
Should we have any logging here, in case the user calls `core.cancel` directly with these things set?